app/containers/RegisterPhone/actions.js

```diff
@@ -79,23 +85,25 @@ export function updateRegisterStatus(status) {
  */
 export function register() {
   return (dispatch, getState) => {
-    const { inputFields } = getState().get('register').toJS();
-
+    const { inputFields } = getState().get('registerPhone').toJS();
     const {
-      first_name, role, email, password, username
+      first_name, email, password, username, social_id, provider, token
     } = inputFields || null;
-
+    let { role } = inputFields || null;
     const { last_name } = inputFields || '';
 
-    if (first_name && role && email && password && username) {
+    if (first_name && role && email && social_id && provider && username && token) {
+      role = Object.keys(ROLES).find(key => ROLES[key] === role)
       DevSummitAxios.post('/auth/register', {
-        first_name, last_name, username, email, password, role
+        first_name, last_name, username, email, password, role, social_id, provider, token
       }).then((response) => {
-        if (response && response.data && response.data.success) {
+        if (response && response.data && response.data.meta && response.data.meta.success) {
           // do something
           dispatch(updateRegisterStatus(true));
+          Alert.alert('Congratulation ' + first_name + ' Registration success (-_-)/ ');
         }
       }).catch((error) => {
+        Alert.alert('Registration failed (T_T)');
```

Comment:
should return some condition to `view` that trigger this alert in the `index` and remove the emoticon btw 😆 @caudaganesh